src/block/utils/Fallable.php

```diff
@@ -33,4 +34,10 @@ interface Fallable{
 	 * Return null if you don't want to change the usual behaviour.
 	 */
 	public function tickFalling() : ?Block;
+
+	/**
+	 * Called just before entity destroys itself.
+	 *
+	 */
+	public function onHitGround(int $fallDistance, int $fallTime, AxisAlignedBB $boundsOnDespawn) : void;
```

Comment:
I would not pass the bounding box here. I would instead replace all parameters with FallingBlock, since you can just get the parameters from there PLUS see which block hit the ground. This should probably also call an event of some sorts. PHPDoc comment too big.